src/test/java/com/example/javaee_jpa_hibernate/model/InvoiceTest.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    @Test
+    public void shouldGetCounterparty() throws Exception {
+        //given
+        Address address = InvoiceGenerator.generateAddress();
+        Counterparty expectedCounterparty = InvoiceGenerator
+                .generateCounterparty("Company name", address, "696 54 24 64",
+                        "123456789", "PKO", "87 1010 1397 0055 0022 2100 0000");
+        Invoice invoice = InvoiceGenerator.generateOneInvoice(LocalDate.now(), expectedCounterparty,
+                InvoiceGenerator.generateInvoiceItems(1));
+        //when
+        Counterparty actualCounterparty = invoice.getCounterparty();
+        //then
+        assertEquals(expectedCounterparty, actualCounterparty);
+        assertEquals(expectedCounterparty.getCompanyName(), actualCounterparty.getCompanyName());
+        assertEquals(expectedCounterparty.getNip(), actualCounterparty.getNip());
+        assertEquals(expectedCounterparty.getAddress(), actualCounterparty.getAddress());
+        assertEquals(expectedCounterparty.getBankName(), actualCounterparty.getBankName());
+        assertEquals(expectedCounterparty.getBankNumber(), actualCounterparty.getBankNumber());
+        assertEquals(expectedCounterparty.getPhoneNumber(), actualCounterparty.getPhoneNumber());
+    }
+
+    @Test
+    public void shouldGetInvoiceItems() throws Exception {
+        //given
+        List<InvoiceItem> expectedInvoiceItem = InvoiceGenerator.generateInvoiceItems(1);
+        Invoice invoice =InvoiceGenerator.generateOneInvoice(LocalDate.now(), InvoiceGenerator.generateCounterparty(),
+                expectedInvoiceItem);
+        //when
+        List<InvoiceItem> actualInvoiceItem = invoice.getInvoiceItems();
+        //then
+        assertEquals(expectedInvoiceItem, actualInvoiceItem);
+        assertEquals(expectedInvoiceItem.get(0), actualInvoiceItem.get(0));
+    }
+
+    @Test
+    public void shouldCheckIfInvoicesAreNotEqual() throws Exception {
+        //given
+        List<Invoice> invoices = InvoiceGenerator
+                .generateListOfInvoices( 2);
+        Boolean expected = true;
+        //when
+        Boolean actual = invoices.get(0).equals(invoices.get(1));
+        //then
+        assertNotEquals(expected, invoices.get(0).equals(null));
+        assertNotEquals(expected, invoices.get(1).equals(null));
+        assertNotEquals(expected, actual);
+    }
+
+    @Test
+    public void shouldReturnHashCode() throws Exception {
+        //given
+        Invoice invoice = InvoiceGenerator
+                .generateOneInvoice();
+        Long id = invoice.getId();
+        int expectedHashCode = 31 * (31 * id.hashCode() + invoice.getDate().hashCode())
+                + invoice.getCounterparty().hashCode();
+        //when
+        int actualHashCode = invoice.hashCode();
+        //then
+        assertEquals(expectedHashCode, actualHashCode);
+    }
+
+    @Test
+    public void shouldCompareInvoices() throws Exception {
```

Comment:
Out of the context question - why do you need an Invoice to implement `Comparable`?